framework/packages/abstract-client/src/source.rs

```diff
@@ -0,0 +1,30 @@
+use abstract_core::objects::{namespace::Namespace, AccountId};
+use cosmwasm_std::Addr;
+
+/// Represents the a route to fetch an account from.
+pub enum AccountSource {
+    /// Get the account from a registered [`Namespace`].
+    Namespace(Namespace),
+    /// Get the account from an [`AccountId`].
+    AccountId(AccountId),
+    /// Get the account from the address of an installed App.
+    App(Addr),
+}
+
+impl From<Namespace> for AccountSource {
+    fn from(namespace: Namespace) -> Self {
+        AccountSource::Namespace(namespace)
+    }
+}
+
+impl From<AccountId> for AccountSource {
+    fn from(account_id: AccountId) -> Self {
+        AccountSource::AccountId(account_id)
+    }
+}
+
+impl From<Addr> for AccountSource {
+    fn from(addr: Addr) -> Self {
+        AccountSource::App(addr)
+    }
+}
```

Comment:
Hm, this one is debatable, because it's not clear what address is that. Should we remove `From<Addr>`, so it's used explicitly `AccountSource::App`?

Reply:
Yeah that makes sense! Will remove it.

Reply:
But then we have to expose the type